SSA/Core/MLIRSyntax/EDSL.lean

```diff
@@ -0,0 +1,57 @@
+import SSA.Core.MLIRSyntax.GenericParser
+import SSA.Core.MLIRSyntax.Transform
+
+/-!
+# MLIR Dialect Domain Specific Language
+This file sets up generic glue meta-code to tie together the generic MLIR parser with the
+`Transform` mechanism, to obtain an easy way to specify a DSL that elaborates into `Com`/`Expr`
+instances for a specific dialect.
+-/
+
+namespace SSA
+
+open Qq Lean Meta Elab Term
+open MLIR.AST
+
+/--
+`elabIntoCom` is a building block for defining a dialect-specific DSL based on the geneeric MLIR
+syntax parser.
+
+For example, if `FooOp` is the type of operations of a "Foo" dialect, we can build a term elaborator
+for this dialect as follows:
+```
+elab "[foo_com| " reg:mlir_region "]" : term => SSA.elabIntoCom reg q(FooOp)
+--     ^^^^^^^                                                        ^^^^^
+```
+-/
+def elabIntoCom (region : TSyntax `mlir_region) (Op : Q(Type)) {Ty : Q(Type)}
+    (_opSignature : Q(OpSignature $Op $Ty) := by exact q(by infer_instance))
+    (φ : Q(Nat) := q(0))
+    (_transformTy      : Q(TransformTy $Op $Ty $φ)     := by exact q(by infer_instance))
+    (_transformExpr    : Q(TransformExpr $Op $Ty $φ)   := by exact q(by infer_instance))
+    (_transformReturn  : Q(TransformReturn $Op $Ty $φ) := by exact q(by infer_instance))
+    :
+    TermElabM Expr := do
+  let ast_stx ← `([mlir_region| $region])
+  let ast ← elabTermEnsuringTypeQ ast_stx q(Region $φ)
+  let com : Q(MLIR.AST.ExceptM $Op (Σ (Γ' : Ctxt $Ty) (ty : $Ty), Com $Op Γ' ty)) :=
+    q(MLIR.AST.mkCom $ast)
+  synthesizeSyntheticMVarsNoPostponing
+  let com : Q(MLIR.AST.ExceptM $Op (Σ (Γ' : Ctxt $Ty) (ty : $Ty), Com $Op Γ' ty)) ←
+    withTheReader Core.Context (fun ctx => { ctx with options := ctx.options.setBool `smartUnfolding false }) do
+      withTransparency (mode := TransparencyMode.all) <|
```

Comment:
Can I have a comment noting that this is terribly aggressive, and in particular, forces us to have the `axiom` trick for FHE?